tools/manifest/item.py

```diff
@@ -205,7 +205,7 @@ def to_json(self):
         if self.jsshell:
             rv[-1]["jsshell"] = True
         if self.script_metadata:
-            rv[-1]["script_metadata"] = self.script_metadata
+            rv[-1]["script_metadata"] = self.script_metadata if PY2 else [(ensure_str(k), ensure_str(v)) for (k,v) in self.script_metadata]
```

Comment:
This doesn't make sense. `ensure_str` on Python 3 is the same as converting to Text. Why aren't we converting to text always (and being explicit in the `ensure` function we choose).

Reply:
I'm not converting to text in py2 just to avoid the extra work. In py2 the json module accepts binary data because it's just str in the end. But in py3 we cannot do that, we need to ensure that we only get str. I don't mind doing the conversion always if you prefer.